File: python/0001_multiples_of_3_and_5.py
# Problem 1: Find the sum of all multiples of 3 and 5 under 1000

import numpy as np


def calc_total(array):
    return np.sum(array)

integers = np.arange(1, 1001)
fizz_buzz_filter = ((integers % 3) == 0) | ((integers % 5) == 0)
to_sum = integers[fizz_buzz_filter]
print("Answer:", calc_total(to_sum))

# %timeit calc_total(to_sum)
# 2.44 µs ± 30.6 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

File: julia/0001_multiples_of_3_and_5.jl
# Problem 1: Find the sum of all the multiples of 3 or 5 below 1000

using BenchmarkTools


function calctotal()

    total = 0

    for i=1:999
        if (i % 3 == 0) || (i % 5 == 0)
            total += i
        end
    end

    total
end


println("Answer:", calctotal())
@benchmark calctotal()
# Answer:233168
# BenchmarkTools.Trial: 
#   memory estimate:  0 bytes
#   allocs estimate:  0
#   --------------
#   minimum time:     1.412 μs (0.00% GC)
#   median time:      1.421 μs (0.00% GC)
#   mean time:        1.437 μs (0.00% GC)
#   maximum time:     12.884 μs (0.00% GC)
#   --------------
#   samples:          10000
#   evals/sample:     10
